### ISSUE TITLE
Incorrect element item namespace handling

### ISSUE THREAD

**Issue by dafanasiev** [mganss/XmlSchemaClassGenerator#214]
See test:
```c#
[Fact]
public void TestArrayOfMsTypeGeneration()
{
    var xsd0 =
        @"<xs:schema xmlns:xs=""http://www.w3.org/2001/XMLSchema"" xmlns:tns=""http://schemas.microsoft.com/2003/10/Serialization/Arrays"" targetNamespace=""http://schemas.microsoft.com/2003/10/Serialization/Arrays"" elementFormDefault=""qualified"">
            <xs:complexType name=""ArrayOfstring"">
                <xs:sequence>
	                <xs:element name=""string"" type=""xs:string"" nillable=""true"" minOccurs=""0"" maxOccurs=""unbounded""/>
                </xs:sequence>
            </xs:complexType>
            <xs:element name=""ArrayOfstring"" type=""tns:ArrayOfstring"" nillable=""true""/>
        </xs:schema>
        ";
    var xsd1 =
        @"<xs:schema xmlns:xs=""http://www.w3.org/2001/XMLSchema"" xmlns:q1=""http://schemas.microsoft.com/2003/10/Serialization/Arrays"" elementFormDefault=""qualified"">
            <xs:import namespace=""http://schemas.microsoft.com/2003/10/Serialization/Arrays""/>
            <xs:complexType name=""c_ai"">
                <xs:sequence>
	                <xs:element name=""d"" type=""q1:ArrayOfstring"" nillable=""true"" minOccurs=""0"">
		                <xs:annotation>
			                <xs:appinfo>
				                <DefaultValue EmitDefaultValue=""false"" xmlns=""http://schemas.microsoft.com/2003/10/Serialization/""/>
			                </xs:appinfo>
		                </xs:annotation>
	                </xs:element>
                </xs:sequence>
            </xs:complexType>
            <xs:element name=""c_ai"" type=""c_ai"" nillable=""true""/>
        </xs:schema>
        ";
    var validXml =
        @"<c_ai xmlns:tns=""http://schemas.microsoft.com/2003/10/Serialization/Arrays"">
            <d>
                <tns:string>String</tns:string>
                <tns:string>String</tns:string>
                <tns:string>String</tns:string>
            </d>
        </c_ai>
        ";
    var generator = new Generator
    {
        IntegerDataType = typeof(int),
        NamespacePrefix = "Test_NS1",
        GenerateNullables = true,
        CollectionType = typeof(System.Collections.Generic.List<>)
    };
    var contents = ConvertXml(nameof(TestArrayOfMsTypeGeneration), new[] {xsd0, xsd1}, generator).ToArray();
    var assembly = Compiler.Compile(nameof(TestForceIsNullableGeneration), contents);
    var testType = assembly.GetType("Test_NS1.C_Ai");
    var serializer = new XmlSerializer(testType);
    Assert.NotNull(serializer);
    dynamic deserialized = serializer.Deserialize(new StringReader(validXml));
    Assert.NotEmpty((IEnumerable)deserialized.D);  //<== oops
}
```

XmlSchemaClassGenerator generate XmlArrayItemAttribute without namespace:
```c#
//...skip
[System.Xml.Serialization.XmlArrayAttribute("d", IsNullable=true)]
[System.Xml.Serialization.XmlArrayItemAttribute("string", IsNullable=true)]  //<==no Namespace
public System.Collections.Generic.List<string> D
{
//....skip
```

xsd.exe generate:
```c#
[System.Xml.Serialization.XmlArrayAttribute(IsNullable=true)]
[System.Xml.Serialization.XmlArrayItemAttribute(Namespace="http://schemas.microsoft.com/2003/10/Serialization/Arrays")]
public string[] d {
....
```

**Comment by mganss** [comment]
@dafanasiev Thanks a lot. Are you planning on making a PR?

**Comment by asvaza** [comment]
Facing same issue. When array item from different namespace as array itself, namespace property on XmlArrayItemAttribute is required. Is there any plans on fixing bug?

**Comment by mganss** [comment]
@asvaza I figured @dafanasiev would already have a PR up his sleeve given that the test was probably more effort than the fix 😄 
I'll procrastinate one more day in the hopes of him jumping in 😉

**Comment by asvaza** [comment]
In case it helps here is my testcase. I use command line "xscgen --verbose -o . Foo.xsd" to generate model files 

XML sample

```xml
<foo:Response xmlns:foo="Foo"
              xmlns:bar="Bar"
              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
              xsi:schemaLocation="Foo foo.xsd Bar bar.xsd">
  <foo:PositionList>
    <bar:Position>
      <bar:Title>baz</bar:Title>
    </bar:Position>
  </foo:PositionList>
</foo:Response>
```

Foo.xsd:
```xml
<xs:schema xmlns="Foo" 
           xmlns:bar="Bar" 
           xmlns:xs="http://www.w3.org/2001/XMLSchema" 
           targetNamespace="Foo" 
           elementFormDefault="qualified" 
           attributeFormDefault="unqualified" 
           version="1.0">
  <xs:import namespace="Bar" schemaLocation="./bar.xsd" />
  <xs:element name="Response" type="ResponseStructure" />
  <xs:complexType name="ResponseStructure">
    <xs:sequence>
      <xs:element name="PositionList" type="bar:PositionListStructure" minOccurs="0" />
    </xs:sequence>
  </xs:complexType>
</xs:schema>
```

Bar.xsd:
```xml
<xs:schema xmlns="Bar" 
           xmlns:xs="http://www.w3.org/2001/XMLSchema" 
           targetNamespace="Bar" 
           elementFormDefault="qualified" 
           attributeFormDefault="unqualified" 
           version="1.0">
  <xs:complexType name="PositionStructure">
    <xs:sequence>
      <xs:element name="Title" type="xs:string" />
   </xs:sequence>
  </xs:complexType>
  <xs:complexType name="PositionListStructure">
    <xs:sequence maxOccurs="unbounded">
      <xs:element name="Position" type="PositionStructure" />
    </xs:sequence>
  </xs:complexType>
</xs:schema>
```

Expected "PositionList" element definition
```c#
        [System.Xml.Serialization.XmlArrayAttribute("PositionList")]
        [System.Xml.Serialization.XmlArrayItemAttribute("Position", Namespace="Bar")]
        public System.Collections.ObjectModel.Collection<Bar.PositionStructure> PositionList
        {
            get
            {
                return this._positionList;
            }
            private set
            {
                this._positionList = value;
            }
        }
```

**Comment by mganss** [comment]
Fixed in 2.0.439.
